unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/AB/AssetPromise_AB.cs

```diff
@@ -61,11 +60,9 @@ protected override void OnCancelLoading()
                 loadCoroutine = null;
             }
 
-            if (assetBundleRequest != null && !assetBundleRequest.isDone)
+            if (asyncOp != null)
```

Comment:
This statement should check if the request is in progress. If we check for null it may not work as expected. (Check out the comment bellow about nullity and running requests)

Reply:
Following the previous explanation, our new `Dispose()` is already checking for the `isDone` param to avoid the abortation or not:

![image](https://user-images.githubusercontent.com/64659061/114909926-940c3400-9e1d-11eb-838f-75a3aa0a30ad.png)

Aparte of that, I think the old code was not disposing the request well. If the request was running, then it was aborted and disposed correctly, BUT if the request was already finished, it was not aborted it (OK, it is not needed) but it didn't dispose either (and it should be disposed always in any cancelation).

Reply:
Cool! On a closer look, if the request is running `asyncOp` should be non-null. Going to resolve this one.